### PR TITLE
On living documentation, initial commit

### PR DESCRIPTION
**Pull request by Zolletta**

### CODE REVIEW

File: content/blog/eng/on-living-documentation.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+# The need for documentation
+Documentation supplements the knowledge we might not have.
+Lack of knowledge manifests in
+- wasted time (finding the missing points or guessing them)
+- biased decisions due to this lack
+
+- Hint: when you don’t know something, you usually don’t know of not knowing ;)
+
+So the time spent on harvesting knowledge should be considered helping build the stakeholders’ application mental model. And this is important because is that model that developers will use to augment the code, product owners will use to describe the stories to implements, and business owners will use to describe their objectives and key results….
```
Comment (BruceGitHub) [inline content/blog/eng/on-living-documentation.md:23]: `stakeholders’` ?

File: content/blog/eng/on-living-documentation.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+By doing this, it hopes to shed a light on the overall design of the application being documented, eventually triggering an opportunity for refactoring.
+
+Overall, it wishes the code to be so clear and self-documented that the additional documentation became useless and redundant.
+
+And that’s why this is not a book about end-user documentation: not every audience is able/willing to read the code, nor that every audience should do it.
+
+#Written documentation is not always the correct answer
+
+The book follows the track of the agile / TDD / DDD / BTD development and builds on their pillars.
```
Comment (BruceGitHub) [inline content/blog/eng/on-living-documentation.md:61]: `BTD` typo

and 

TDD | BDD DDD? 
Test first

File: content/blog/eng/on-living-documentation.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+> With TDD, the tests are first considered as specifications. With DDD, we identify the code and the model of the business domain […]. … we expect the code to tell the whole story about the domain.
```
Comment (BruceGitHub) [inline content/blog/eng/on-living-documentation.md:63]: with TDD and BDD

Comment (Zolletta) [reply content/blog/eng/on-living-documentation.md:63]: DDD che cosa era nella mia testa? un Developer-Driven-Development ?? LOL

File: content/blog/eng/on-living-documentation.md [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+
+It touches code organization (aka, in which folder should I put this kind of files?), refactoring, architecture, agile, test-driven development…
+
+I think it touches a lot of the aspects that make software development a craftmanship.
+
+And I think several kinds of audiences would benefit from reading this book.
+
+Besides developers, I think product owners would benefit because, apart from answering the questions "why we should document?' and "what?", it gives hints on how to write stories on Jira (which are indeed, some kind of documentation too).
+
+I think some aspects touched in the book (decision logs, for instance) could be applied to any level of a company, of course with different kinds of granularity.
```
Comment (mar-gil) [inline content/blog/eng/on-living-documentation.md:190]: I'd write "discussed in the book"

**Review by mar-gil — COMMENTED** [review]
Here's my review :) I am available for any clarification.

File: content/blog/eng/on-living-documentation.md [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+
+I admit: I am (sometimes) dedicated to the antipattern: human dedication (see page 32).
+
+Sometimes I need to sketch up things while following the execution of the code, and it’s a pity to throw away those findings.
+
+Other times I need to clear up my mind over some convoluted code.
+
+In any case, I resort to the tool that the company I work for offers: Confluence pages.
+
+Once I’ve done writing the article, that article begins to age, eventually reaching, as the author says, the information graveyard.
```
[the remark this answers is not in the document]
Comment (Zolletta) [reply content/blog/eng/on-living-documentation.md:39]: Ok, ma lascio information graveyard come dice l'autore.

File: content/blog/eng/on-living-documentation.md [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+> 
+> <cite>Source: the book, page 12</cite>
+
+Since the purpose of the documentation is a transfer of knowledge, in many cases the preferable and more convenient way to achieve this is by having conversations, which is another way of saying
+
+> The most efficient and effective method of conveying information to and within a development team is face-to-face conversation.
+>
+> <cite>>Source: the Agile Manifesto, https://agilemanifesto.org/iso/en/principles.html**
+
+Fact is that even when we favour “conversation over documentation”, sometimes we need to summarize that we have discovered. That summary will be less wordy and up-to-the-point since all the assumptions and agreements have already been discussed.
```
[the remark this answers is not in the document]
Comment (Zolletta) [reply content/blog/eng/on-living-documentation.md:73]: yes typo!

File: content/blog/eng/on-living-documentation.md [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+
+- Tests written for behavior-driven-development
+- Configuration files
+- Some README.md that you might have written times ago
+- Annotations you already use for documenting routes, or for your ORM (like Doctrine) …
+- …
+
+The point is that every kind of information that “decorates” the code, gives it some context, and functions as a way to transfer knowledge (even from yourself of today to yourself of the future) is documentation.
+
+Moreover, since most of this stuff needs always to be up-to-date because it is needed by the application to run (or to deploy), you can be assured that it will be always accurate and can become the single source of truth.
```
[the remark this answers is not in the document]
Comment (Zolletta) [reply content/blog/eng/on-living-documentation.md:105]: ok ma lascio up-to-date (intendevo qualcosa come "sempre sul pezzo" non aggiornata)

File: content/blog/eng/on-living-documentation.md [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+
+If anything, the problem of this kind of knowledge is its fragmentation and / or obfuscation. Not to mention the worst case scenario, that that knowledge is only in the developer’s mind and in the code we can read only its consequences.
+
+> Documentation accelerates delivery because it shortens the time to rebuild your mental model of the system to work on.
+> 
+> <cite>Source: book, page 341</cite>
+
+#Internal vs external documentation
+
+The author distinguishes between internal documentation (placed in the code, for example via annotations or codes or near it, i.e. with README.md files) and external documentation. This kind of documentation can be leveraged even with the most sophisticated IDEs.
```
[the remark this answers is not in the document]
Comment (Zolletta) [reply content/blog/eng/on-living-documentation.md:115]: oops... ho messo "The former can be leveraged even with the most sophisticated IDEs."

File: content/blog/eng/on-living-documentation.md [hunk 9]
```diff
@@ -0,0 +1,194 @@
...
+The author defines such kind of documentation “evergreen documentation” (see p. 246).
+
+#Some guidelines and suggestions
+
+- Only one single source of truth is allowed: if you need to make it accessible somewhere else, use references (i.e. links or annotations that, in the publication phase, will copy from the source on the destination).
+- Of course, with every publication, we provide an up-to-date snapshot of what was current at the time of the information harvesting. That’s why we should clearly version the documentation every time we publish it.
+- Every published document should be considered read-only: if you need to amend it, publish another version.
+- Use annotations to augment your code: the code tells the how, the annotations could tell the why, aka the rationale (see p. 122)
+- Annotations can be used to mark security issues (@securityIssue(name=’blaBla’, jira=’<link to jira>’), or to declare that a class should have not setters
+- Stick to one of these styles: declare @ Immutable, @NotNullable vs @ Mutable and @ Nullable. Don’t mix styles to avoid inconsistencies. [I put a space between @ and the annotation not to tag anyone ]
```
[the remark this answers is not in the document]
Comment (Zolletta) [reply content/blog/eng/on-living-documentation.md:165]: Actually is the other way: @Immutable is correct :)

File: content/blog/eng/on-living-documentation.md [hunk 10]
```diff
@@ -0,0 +1,194 @@
...
+- You could create guided tours of particular sections of your code
+- You could integrated also some diagrams (see **[Graphviz](https://graphviz.org/)** and “Living diagrams” and following, page 170).
+- Don’t use primitive types, create your own: types don’t lie, and if you don’t use them properly, the pipeline will be red!
+- Use tools to enforce the decisions made into guidelines (p. 301): they provide knowledge just in time where is needed (when someone violets them).
+
+#Conclusion
+
+That's (a lot) more than living documentation inside this book.
+
+It goes on talking about code quality and code design techniques: for example, even when naming a method we could introduce a burden on the shoulders of the next developer or alleviate her voyage.
```
[the remark this answers is not in the document]
Comment (Zolletta) [reply content/blog/eng/on-living-documentation.md:180]: To answer your question: yes :)

**Pull request merged**